addons/binding/org.openhab.binding.geofence/README.md

```diff
@@ -0,0 +1,160 @@
+        "location": "xx.xxxx,yy.yyyy",
+        "radius": 100,
+        "triggerEvent": false
+    },
+    ...
+ ]
+ ```
+
+### OwnTracks
+
+Install [OwnTracks for Android](https://play.google.com/store/apps/details?id=org.openhab.habdroid) or [OwnTracks for iOS](https://itunes.apple.com/us/app/owntracks/id692424691) on your device.
+
+Go to Preferences/Connection and set:
+
+* **Mode** - select Private HTTP
+* **Host** - **https://<your.ip.address>/geofence/owntracks**
+* **Identification**
+  * Turn Authentication ON
+  * Set username and password to be able to reach your openHAB server
+  * Device ID is not important. Set it to e.g. phone
+  * Tracker ID - This id will identify the tracker as a thing. I use initials here.
+  
+![Image](./docs/owntracks_setup.png)
+
+### GPSLogger
+
+Install [GPSLogger for Android](https://play.google.com/store/apps/details?id=com.mendhak.gpslogger) on your device. After the launch, go to General Options. Enable Start on bootup and Start on app launch.
+
+![Image](./docs/gpslogger_1.png)
+
+Go to Logging details and disable Log to GPX, Log to KML and Log to NMEA. Enable Log to custom URL.
+
+![Image](./docs/gpslogger_2.png)
+
+Right after enabling, the app will take you to the Log to custom URL settings and modify:
+
+* **URL** - Set URL to **https://<username>:<password>@<your.ip.address>/geofence/gpslogger** by replacing <???> with values matching your setup. It’s HIGHLY recommended to use SSL/TLS.
+* **HTTP Body** - Type in the following JSON:
+
+```
+{
+    "_type":"location",
+    "lat":%LAT,
+    "lon":%LON,
+    "tid":"XY",
+    "acc":%ACC,
+    "batt":%BATT,
+    "tst":%TIMESTAMP,
+    "alt":%ALT,
+    "cog":%DIR,
+    "vel":%SPD
+}
+```
+
+**Note**: The value of "tid" is the tracker identified that will identify the tracker as a thing in openHAB. This must be unique for each tracker connected to the same openHAB instance (e.g. family members).
+
+* **HTTP Method** - type in: **POST**
+* **HTTP Headers** - type in: **Content-Type: application/json**
+
+![Image](./docs/gpslogger_3.png)
+
+### Things
+
+It is possible to define things manually or to use the discovery feature of the openHAB. An important detail for both methods is that a tracker is identified by a **tracker id** configured on mobile devices. Make sure these tracker ids are unique in group of trackers connected to a single openHAB instance.
+
+#### Manual setup
+
+```
+//phones
+Thing geofence:device:XY   "XY tracker" [ipAddress="10.200.0.106", ipAddressStatic=true, pingInterval=15]
+Thing geofence:device:PQ   "PQ tracker"
+...
+```
+
+**Note**: By defining a static IP address for a device the binding will start checking the presence of the device on the local network with the given ping interval.
+
+#### Discovery
+
+If the things are not defined in **.things** files the first time the device sends a GPS log record the binding will recognize it as new device and insert an entry into the Inbox as new device with name **GPS Tracker:??**.
```

Comment:
This does not work automatically (in the background) - I only see these results when triggering a manual discovery afterwards.

Reply:
It works for me. Could you check in the new build? The first GPS log record adds the device into the inbox. After creating the new thing from the inbox the thing stays in INITIALIZING state until the second GPS log record, when it changes to ONLINE.

Reply:
Yes, because you have meanwhile added a `thingDiscovered` call in the `addTracker` method, which was previously not the case.

Reply:
> After creating the new thing from the inbox the thing stays in INITIALIZING state

This is a bug - the Thing should be set to UNKOWN until you know that it is ONLINE (and probably go to OFFLINE, if you did not receive any reports for a certain amount of time).

Reply:
Ok. I'll set UNKOWN in initialize. But it is not that obvious when you need to set the thing to OFFLINE. If somebody stays home for 3 days there will be no GPS log records for that time. What shall I do in this case?

Reply:
I agree that it is difficult to decide when to go OFFLINE (btw, this does not only concern the transition UNKNOWN->OFFLINE, but also ONLINE->OFFLINE).
So there is no maximum interval in which the trackers would send an update? Would be good to have something like a kind of still-alive ping from them. If it can be a normal situation that there is no incoming record for days then I'd think we can simply always set the status to ONLINE, no matter what.

Reply:
> we can simply always set the status to ONLINE, no matter what.

This is the case now. After the second record it is set to ONLINE and it stays there...

Reply:
Why only after the second record? It could be set to ONLINE in initialize right away.